components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -377,19 +377,19 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
       let notifications: Record<number, RewardsExtension.Notification> = state.notifications
       let id = ''
 
-      if (!notifications) {
-        notifications = []
+      if (!notifications || Array.isArray(notifications)) {
+        notifications = {}
       }
 
-      list.forEach((notification: RewardsExtension.Notification) => {
-        id = notification.id
-        notifications[notification.id] = {
-          id: notification.id,
-          type: notification.type,
-          timestamp: notification.timestamp,
-          args: notification.args
+      for (const notificationItemData of list) {
```

Comment:
not sure if this change is needed

Reply:
Correct, this was just cleanup to aid readability when diagnosing the bug

Reply:
could we revert back to what it was so that change is really minimal as we will probably need to uplift it into beta

Reply:
++ on revert back, just to minimize patch (even though it's arguably nicer; I always avoid `forEach` since you can't break out)

Reply:
Actually I was wondering if this was part of the bug in case payload.list is actually the same item in memory as state.notifications, and then it was modifying the list that was being iterating over.... I don't think so though if it's _always_ coming from c++, but I can't vouch for that. I'll revert this, but I did check that this is unchanged down to at least 0.65, and it is unchanged...